### PR TITLE
Deb Package Targets

### PR DESCRIPTION
**Pull request by brthor**
@eerhardt @livarcocc @sokket @piotrpMSFT

### CODE REVIEW

File: build/package/Microsoft.DotNet.Cli.Nupkg.targets [hunk 1]
```diff
@@ -59,7 +59,7 @@
           Inputs="@(ProjectPackTargetInputs -> '%(Identity)')"
           Outputs="@(ProjectPackTargetOutputs -> '%(Identity)')">
     
-    <Exec Command="dotnet pack %(ProjectsToPack.Identity) 
+    <Exec Command="$(DotnetStage2) pack %(ProjectsToPack.Identity)
```
Comment (eerhardt) [inline build/package/Microsoft.DotNet.Cli.Nupkg.targets]: Can this use the DotNetPack ToolTask Peter just added instead?

File: scripts/package/package-debian.sh [hunk 2]
```diff
@@ -115,10 +115,6 @@ parseargs(){
         echo "Provide the NuGet name of the targetted Shared Framework. Missing option '--framework-nuget-name'" && help
     fi
 
-    if [ -z "$PREVIOUS_VERSION_URL" ]; then
```
Comment (eerhardt) [inline scripts/package/package-debian.sh]: I see this is commented out below.  What's the plan for enabling upgrade tests?  Do we have an issue for it?  If not, can you log one?

File: build/package/Microsoft.DotNet.Cli.Installer.DEB.targets [hunk 3]
```diff
@@ -1,9 +1,114 @@
...
+
+        <!-- Rebuild when version changes -->
+        <MakeDir Directories="$(SdkDebianIntermediateDirectory)" />
+        <WriteLinesToFile
+          File="$(SdkVersionSentinel)"
+          Lines="$(SdkVersion)"
+          Condition= "!Exists('$(SdkVersionSentinel)')"
+          Encoding="Unicode"/>
+
+        <Message Condition=" '$(DebuildPresent)'  != 'true' "
```
Comment (eerhardt) [inline build/package/Microsoft.DotNet.Cli.Installer.DEB.targets]: Should this be an error?

Comment (eerhardt) [inline build/package/Microsoft.DotNet.Cli.Installer.DEB.targets]: Or possibly a warning?

---
In reply to: [70663371](https://github.com/dotnet/cli/pull/3845#discussion_r70663371) [](ancestors = 70663371)

File: scripts/package/package-debian.sh (hunk 2 above)
Comment (brthor) [inline scripts/package/package-debian.sh]: Upgrade tests are a relic from pre-shared fx. We don't have an upgrade story any more :) just SxS

File: build/package/Microsoft.DotNet.Cli.Installer.DEB.targets (hunk 3 above)
Comment (brthor) [inline build/package/Microsoft.DotNet.Cli.Installer.DEB.targets]: The Importance high was intended to signify a warning, but no it shouldn't be an error. I'll add back the comment that htis is a workaround for ubuntu 16 not present on jenkins

File: build/package/Microsoft.DotNet.Cli.Installer.DEB.targets [hunk 4]
```diff
@@ -1,9 +1,114 @@
...
+            <SdkVersionSentinel>$(SdkDebianIntermediateDirectory)/$(SdkVersion).sentinel</SdkVersionSentinel>
+        </PropertyGroup>
+
+        <ItemGroup>
+            <SdkDebInputFiles Include="$(SdkLayoutOutputDirectory)/**/*" />
+        </ItemGroup>
+
+        <!-- Rebuild when version changes -->
+        <MakeDir Directories="$(SdkDebianIntermediateDirectory)" />
+        <WriteLinesToFile
```
Comment (eerhardt) [inline build/package/Microsoft.DotNet.Cli.Installer.DEB.targets]: Do we have an existing "version" file that we can use instead of making a one-off here?

Comment (eerhardt) [inline build/package/Microsoft.DotNet.Cli.Installer.DEB.targets]: like maybe the .version file in sdk/<version>

---
In reply to: [70674044](https://github.com/dotnet/cli/pull/3845#discussion_r70674044) [](ancestors = 70674044)

Comment (brthor) [inline build/package/Microsoft.DotNet.Cli.Installer.DEB.targets]: I'm going to take this out for the moment, we're not doing this for the other installers, and I want to think through this further.

File: build/package/Microsoft.DotNet.Cli.Nupkg.targets (hunk 1 above)
Comment (brthor) [inline build/package/Microsoft.DotNet.Cli.Nupkg.targets]: Fixed.

File: scripts/package/package-debian.sh (hunk 2 above)
Comment (eerhardt) [inline scripts/package/package-debian.sh]: Ok, can we remove the commented out section below as well then?  So it is completely out of the repo?

Comment (eerhardt) [inline scripts/package/package-debian.sh]: Specifically this part:

```
     # Set LAST_VERSION_URL to enable upgrade tests
     # Temporarily disable last version until we have one with shared fx
     # export LAST_VERSION_URL="$PREVIOUS_VERSION_URL"
```

Comment (brthor) [inline scripts/package/package-debian.sh]: fixed.

Comment (eerhardt) [inline scripts/package/package-debian.sh]: Thanks!

**Comment by eerhardt** [comment]
:shipit:

**Pull request merged**